fdb-extensions/src/test/java/com/apple/foundationdb/async/rabitq/RaBitQuantizerTest.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * RaBitQuantizerTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.async.rabitq;
+
+import com.apple.foundationdb.linear.DoubleRealVector;
+import com.apple.foundationdb.linear.FhtKacRotator;
+import com.apple.foundationdb.linear.Metrics;
+import com.apple.foundationdb.linear.RealVector;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ObjectArrays;
+import com.google.common.collect.Sets;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public class RaBitQuantizerTest {
+    private static final Logger logger = LoggerFactory.getLogger(RaBitQuantizerTest.class);
+
+    @Nonnull
+    private static Stream<Arguments> randomSeedsWithDimensionalityAndNumExBits() {
+        return Sets.cartesianProduct(ImmutableSet.of(3, 5, 10, 128, 768, 1000),
+                        ImmutableSet.of(1, 2, 3, 4, 5, 6, 7, 8))
+                .stream()
+                .flatMap(arguments ->
+                        LongStream.generate(() -> new Random().nextLong())
+                                .limit(3)
+                                .mapToObj(seed -> Arguments.of(ObjectArrays.concat(seed, arguments.toArray()))));
+    }
+
+    @Test
+    void basicEncodeTest() {
+        final int dims = 768;
+        final Random random = new Random(System.nanoTime());
```

Comment:
Should this be parameterized by seed? There are a lot of these in this class, so maybe what we really need to do is parameterize the class by seed